core/deployment/src/main/java/io/quarkus/deployment/dev/LauncherMain.java

```diff
@@ -0,0 +1,38 @@
+package io.quarkus.deployment.dev;
+
+import java.io.File;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+
+import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
+
+import io.smallrye.config.SmallRyeConfigProviderResolver;
+
+/**
+ * The entry point when launched from an IDE
+ */
+public class LauncherMain {
+
+    public static void main(Path appClasses, Path wiring, URL[] classPath, String... args) throws Exception {
+        DevModeContext context = new DevModeContext();
+        context.setAbortOnFailedStart(true);
+        context.setTest(false);
+        context.setCacheDir(Files.createTempDirectory("quarkus-cache").toFile());
+        context.setSourceEncoding("UTF-8");
+        File appClassesFile = appClasses.toFile();
+        context.getClassesRoots().add(appClassesFile);
+
+        //TODO: huge hacks
+        File src = new File(appClassesFile, "../../src/main/java");
```

Comment:
good enough to start, but I reckon just doing a recursive search for nearest pom.xml/build.gradle and resolve source paths from there (still just via filesystem not parsing the build files) could make this a bit less hacky.

Reply:
Also this would fail on windows because of the delimiters, or?

Reply:
Forward slash is also a delimiter on windows.

Reply:
That's good to know, I always thought it's only `\` on windows... ;-)